content/deyemiobaa/building-a-todolist-with-rails.md

```diff
@@ -0,0 +1,355 @@
+
+In order to display the list of todos, we need to get the list of todos from the database. Open the `app/controllers/todos_controller.rb` and add the following code to the `index` method.
+
+```ruby
+...
+def index
+  @todos = Todo.all
+end
+...
+```
+
+The `Todo.all` method returns an array of all the Todo models in the database. The `@todos` variable is now available in the view.
+Once you have the code above, you should see a table with the list of todos previous created on the browser.
+
+## Adding a delete button
+
+Let's add a functionality to delete a todo item. We'll add a delete button to each row in the table. Open the `app/views/todos/index.html.erb` and add an extra column to the table head and table body with the following code.
+
+```erb
+...
+  <thead>
+    <tr>
+      <th class="p-3 text-sm font-semibold text-left text-gray-700 bg-gray-200">ID</th>
+      <th class="p-3 text-sm font-semibold text-left text-gray-700 bg-gray-200">Description</th>
+      <th class="p-3 text-sm font-semibold text-left text-gray-700 bg-gray-200">Action</th>
+    </tr>
+  </thead>
+
+  
+  <tbody>
+    <% @todos.each do |todo| %>
+      <tr>
+        <td class="px-3 py-4 text-sm font-normal text-left text-gray-600 bg-gray-50">
+          <%= todo.id %>
+        </td>
+        <td class="px-3 py-4 text-sm font-normal text-left text-gray-600 bg-gray-50">
+          <%= todo.description %>
+        </td>
+        <td class="px-3 py-4 text-sm font-normal text-left text-red-600 underline break-words bg-light">
+          <%= button_to "Delete", todo_path(todo), method: :delete %>
+        </td>
+      </tr>
+    <% end %>
+  </tbody>
+```
+
+The `button_to` method creates a form with a button. The `method: :delete` option specifies that the form should be submitted using the `DELETE` HTTP verb. The `todo_path(todo)` syntax generates the path for the todo. The `todo_path` method takes the todo model as an argument.
+
+Next, we need to add a route to handle the delete request. Open the `config/routes.rb` file and update the todos resource with the following code.
+
+```ruby
+...
+  resources :todos, only: [:index, :create, :destroy]
+```
+
+The `destroy` method will handle the delete request. 
+
+Open the `app/controllers/todos_controller.rb` file and add the following code to create `destroy` method.
+
+```ruby
+...
+  def destroy
+    @todo = Todo.find(params[:id])
+    @todo.destroy
+    render :index
+  end
+```
+
+The `Todo.find` method finds a todo by the id. Every todo gets a unique id when it's created. The `destroy` method deletes the todo from the database. The `render :index` method renders the `index` view.
+
+Now, you should be able to delete a todo item from the list.
+
+That is all. We’ve built a working todo list with READ, CREATE and DESTROY methods.
+
+You can find the working code on this [Github repo](https://github.com/deyemiobaa/Rails_todolist).
```

Comment:
I've gotten through the bulk of the tutorial and having an error pop up on the last little bit. When I click on the "Delete" button for a todo, I'm getting an error:

![image](https://user-images.githubusercontent.com/86081858/234677721-8b86a4ae-3ab4-4565-a63d-e36be3f57345.png)

It seems to be on rendering the page after a delete, as the delete itself works fine. Refreshing gets rid of the error.

It also happens when I submit a todo again. (It wasn't happening when I added a bunch of todos before making the table and the delete buttons.)